sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-factory.ts

```diff
@@ -0,0 +1,340 @@
+  DisbursementValue,
+  FullTimeAssessment,
+  InstitutionLocation,
+  ProgramYear,
+  Student,
+  StudentAssessment,
+} from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createFakeInstitutionLocation,
+  createFakeStudent,
+  createFakeUser,
+  ensureProgramYearExists,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
+import {
+  IER12Application,
+  IER12Assessment,
+  IER12Disbursement,
+  IER12Offering,
+  IER12Program,
+  IER12Student,
+  IER12TestInputData,
+} from "./models/data-inputs";
+import { addDays, dateDifference, getISODateOnlyString } from "@sims/utilities";
+import * as faker from "faker";
+
+/**
+ * Save all IER12 related records providing all data that
+ * will be part of the IER12 file generation.
+ * @param db data source helper.
+ * @param testInputData IER12 test input data.
+ * @param options method options:
+ * - `programYearPrefix` program year prefix to create or find the program year to
+ * be associated. Default 2000, would will create the program year 2000-2001.
+ * - `submittedDate` date that the application was submitted.
+ * @returns a saved application with all related data needed for an IER12 record creation.
+ */
+export async function saveIER12TestInputData(
+  db: E2EDataSources,
+  testInputData: IER12TestInputData,
+  options?: {
+    programYearPrefix?: number;
+    submittedDate?: Date;
+  },
+): Promise<Application> {
+  // If a program year prefix is not provided, create one with the year 2000 as default.
+  const programYear = await ensureProgramYearExists(
+    db,
+    options?.programYearPrefix ?? 2000,
+  );
+  // Submission date to be used as base date to non-program-year related dates.
+  const referenceSubmission = options?.submittedDate ?? new Date();
+  // Supporting shared variables.
+  const createSecondDisbursement =
+    testInputData.assessment.disbursementSchedules.length === 2;
+  // Set the number of weeks to some number beyond 17 if there are two disbursements
+  // since two disbursements are supposed to be create to offerings longer than 17 weeks.
+  const offeringEndDateWeeks = createSecondDisbursement ? 18 : 10;
+  // Set offering start date to few days after the program year start date.
+  const studyStartDate =
+    testInputData.offering.studyStartDate ??
+    getISODateOnlyString(addDays(15, programYear.startDate));
+  // Set the offering end date to few weeks after the program year start date.
+  const studyEndDate =
+    testInputData.offering.studyEndDate ??
+    getISODateOnlyString(
+      addDays(7 * offeringEndDateWeeks, programYear.startDate),
+    );
+  // User, Student, and SIN validation.
+  const student = await saveIER12StudentFromTestInput(
+    db,
+    testInputData.student,
+  );
+  // Location
+  const institutionLocation = createFakeInstitutionLocation();
+  institutionLocation.institutionCode = "ZZZY";
```

Comment:
Minor. ZZZY could be constant

Reply:
It can be, but taking a look at the factories we do assign the values directly almost all the time, I just did not see a benefit here. It would be a different conversation if it was a production code.